### DISCUSSION TITLE
Quarto  Render on Save, Visual Editor Button, move to parent Folder in explorer

### DISCUSSION THREAD

**Discussion by timonscheuer** [posit-dev/positron#4640] [Feedback]
Hey!

as a heavy Quarto user, I'd like to request a "Render on Save"-opportunity like in RStudio. Makes it so much easier to quickly check the output. Secondly, I'd be very happy for an R-Studio-like Button to quickly switch between source code and Visual Editor. Thirdly, I'd love to  have an option in the Explorer window that allows me to navigate not only to subfolders but also back to the parent folder of my start folder or working directory, so I can move both deeper and higher in the directory structure.

Thank you! :)

**Comment by jennybc** [comment]
Your Quarto experience inside Positron is coming from the Quarto extension. Its GitHub home is here: <https://github.com/quarto-dev/quarto/tree/main/apps/vscode>.

I see a section there on how to configure "Render on Save" for various scopes (e.g. user-level, project, document). Does this answer your question?

https://github.com/quarto-dev/quarto/tree/main/apps/vscode#render-on-save

As for a button to toggle between seeing source vs. the visual editor: I assume you've found the menu item available from the `...` in the upper right hand corner (but just wanted to check). I can't find an existing issue about making that available via a button, so I'd invite you to request that feature on this repo:

https://github.com/quarto-dev/quarto

Finally about this:

> an option in the Explorer window that allows me to navigate not only to subfolders but also back to the parent folder of my start folder or working directory, so I can move both deeper and higher in the directory structure

With the caveat that I might be misunderstanding you, I think this is currently a "no, won't happen" for at least two reasons. First, a big motivator for building Positron on top of VS Code is to get basic IDE features from VS Code and then concentrate on data science specific features in Positron. I don't think this request fits the bill. And second, I basically think the aggressive stance in VS Code / Positron on focusing on the current workspace does a lot more good than harm. So I doubt that freeing the file explorer to start browsing "higher" in the hierarchy than the currently-opened workspace is something we would prioritize.

That being said, you can't be the first person to have this wish. So it's possible you can find a VS Code extension that tweaks the explorer in this way.

(There do seem to be lots of interesting commands in the command palette around the explorer, so I'd encourage you to peruse those.)
